public/js/components/SourceSearch.js

```diff
@@ -61,27 +57,27 @@ const Search = React.createClass({
 
   toggle(key, e) {
     e.preventDefault();
-    this.setState({ searchOn: !this.state.searchOn });
+    this.props.toggleSearch(!this.props.searchOn);
   },
 
   onEscape(shortcut, e) {
-    if (this.state.searchOn) {
+    if (this.props.searchOn) {
       e.preventDefault();
-      this.setState({ searchOn: false });
+      this.props.toggleSearch(false);
     }
   },
 
   close() {
-    this.setState({ searchOn: false });
+    this.props.toggleSearch(false);
   },
 
   render() {
-    return this.state.searchOn ?
+    return this.props.searchOn ?
       dom.div({ className: "search-container" },
       Autocomplete({
         selectItem: result => {
           this.props.selectSource(result.id);
-          this.setState({ searchOn: false });
+          this.props.toggleSearch(false);
```

Comment:
Hmm, lets see what the reducer option looks like. We do want a UI state reducer at some point. 

I'm on the fence about the DOM event. I think i'd also feel differently if instead of it being a keydown it were clicking an element :)